webui/react/src/services/stream/keyCache.ts

```diff
@@ -0,0 +1,86 @@
+export function decode_keys(keys: string): Array<number> {
+  const retval: Array<number> = [];
+  if (keys.length === 0) {
+    return retval;
+  }
+  for (const key of keys.split(',')) {
+    if (key.includes('-')) {
+      const [start, end] = key.split('-');
+      for (let i = Number(start); i <= Number(end); i++) {
+        retval.push(i);
+      }
+    } else {
+      retval.push(Number(key));
+    }
```

Comment:
tiny nit -- consider instead:
```suggestion
    const [start, end] = key.split('-', 2);
    if (end) {
      for (let i = Number(start); i <= Number(end); i++) {
        retval.push(i)
      }
    } else {
      retval.push(Number(key))
    }
```
this does two things:
1) instead of checking before we split, we split and check if there were the expected amount of components
2) we guarantee the string is split into 2 components maximum